nexteuropa_language_selector_site_popup/nexteuropa_language_selector_site_popup.module

```diff
@@ -47,6 +47,11 @@ function nexteuropa_language_selector_site_popup_preprocess_splash(&$variables)
     if (strpos($href, 'node/') == 0 &&
         function_exists('translation_path_get_translations')) {
       $translations = translation_path_get_translations($href);
+      if (preg_match('/(node)\/(\d+)/', $_GET['destination'], $matches)) {
```

Comment:
This condition looks pretty redundant...
you have a first condition about the preg_match returned value, this condition is enough, i would say, if it returns true that also means that $matches is populated and it has three values, the second is "node" (you query looks only for that) , the third is "numeric" (same as before), no reason for checking further, basically.